src/react/components/pages/editorPage/canvas.tsx

```diff
@@ -351,38 +376,87 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
+                    this.playerRef.current.manager.rootElement.style.zIndex = 0;
+                    this.editor.addContentSource(this.playerRef.current.video.video);
+                    this.updateRegions();
+                } else if (!state.paused) {
+                    // We need to make sure we're on top if we are playing
+                    this.playerRef.current.manager.rootElement.style.zIndex = 9001;
+                }
+            });
+        } else {
+            // Something has gone majorly wrong to get to this spot
+            throw new Error(strings.errors.unknown.message);
+        }
+    }
+
+    /**
+     * @name loadImage
+     * @description loads an image into the canvas
+     * @returns {void}
+     */
+    private loadImage() {
         const image = new Image();
         image.addEventListener("load", (e) => {
-            console.log("loading");
             // @ts-ignore
             this.editor.addContentSource(e.target);
-            if (this.props.selectedAsset.regions.length) {
-                this.props.selectedAsset.regions.forEach((region: IRegion) => {
-                    const loadedRegionData = new RegionData(region.boundingBox.left,
-                                                            region.boundingBox.top,
-                                                            region.boundingBox.width,
-                                                            region.boundingBox.height,
-                                                            region.points.map((point) => new Point2D(point.x, point.y)),
-                                                            this.regionTypeToType(region.type));
-                    if (region.tags.length) {
-                        this.addRegion(region.id, this.scaleRegionToFrameSize(loadedRegionData),
-                                        new TagsDescriptor(region.tags.map((tag) => new Tag(tag.name,
-                                            this.props.project.tags.find((t) => t.name === tag.name).color))));
-                    } else {
-                        this.addRegion(region.id, this.scaleRegionToFrameSize(loadedRegionData), new TagsDescriptor());
-                    }
-                    if (this.state.selectedRegions) {
-                        this.setState({
-                            selectedRegions: [this.props.selectedAsset.regions[
-                                this.props.selectedAsset.regions.length - 1]],
-                        });
-                    }
-                });
-            }
+            this.updateRegions();
         });
         image.src = this.props.selectedAsset.asset.path;
     }
 
+    private updateRegions() {
+        if (this.props.selectedAsset.regions.length) {
+            this.props.selectedAsset.regions.forEach((region: IRegion) => {
+                const loadedRegionData = new RegionData(region.boundingBox.left,
```

Comment:
I know this isn't your code - but would be a good idea to make a generic transform function that can covert a `IRegion` from our app model into the `RegionData` instance that canvas tool expects.